marahgh/assignment1/Q1.cpp

```diff
@@ -0,0 +1,129 @@
+
+#include <iostream>
+#define LEN 'z' -'a' +1
+using namespace std;
+
+/*anagram words */
+//My assumption:the words alphabet is english letters.    
+
+
+
+char lowerIt(char c){
+    if(c>='a' && c<='z'){
+        return c;
+    }
+    return 'a' + c-'A';
+}
+//---------------------------------------------------------------------------------------------------------
+/*A function that determines if two strings are anagrams ,
+	able to handle both case sensitive and case insensitive anagrams */
+bool areAnagrams(string s1, string s2){
+    if(s1.size()!= s2.size()){
+        return false;
+    }
+    int histogram[LEN];
+    for(int i=0;i<LEN;i++){
+        histogram[i]=0;
+    }
+    for(int i=0;i<s1.size();i++){

```

Comment:
it may be more readable to use the new style iterators: `for (char c : s1) { ... } `